--- /usr/src/ixgbevf-4.3.4/src/kcompat.h.orig	2018-03-14 12:39:07.125424467 +0000
+++ /usr/src/ixgbevf-4.3.4/src/kcompat.h	2018-03-14 12:54:41.989425650 +0000
@@ -22,6 +22,10 @@
 
 *******************************************************************************/
 
+#define OPENSUSE_423 /* The kernel in openSUSE 42.3 includes some patches which 
+                        are also part of this file and thus need to be 
+                        disabled. */
+
 #ifndef _KCOMPAT_H_
 #define _KCOMPAT_H_
 
@@ -5410,6 +5414,7 @@
 }
 #endif
 
+#ifndef OPENSUSE_423
 #if !(UBUNTU_VERSION_CODE && \
 		UBUNTU_VERSION_CODE >= UBUNTU_VERSION(4,4,0,21)) && \
 	!(RHEL_RELEASE_CODE && \
@@ -5429,6 +5434,7 @@
 	* sum = csum_fold(csum_add(diff, ~csum_unfold(*sum)));
 }
 #endif
+#endif /* !OPENSUSE_423 */
 
 #if !(RHEL_RELEASE_CODE && (RHEL_RELEASE_CODE > RHEL_RELEASE_VERSION(7,2)))
 static inline void page_ref_inc(struct page *page)
@@ -5471,6 +5477,7 @@
 };
 #endif
 
+#ifndef OPENSUSE_423
 #if !(SLE_VERSION_CODE && (SLE_VERSION_CODE >= SLE_VERSION(12,3,0))) &&\
 	!(RHEL_RELEASE_CODE && RHEL_RELEASE_CODE >= RHEL_RELEASE_VERSION(7,4))
 static inline int
@@ -5509,6 +5516,7 @@
 			    pci_select_bars(pdev, IORESOURCE_MEM));
 }
 #endif /* !SLE_VERSION(12,3,0) */
+#endif /* !OPENSUSE_423 */
 #else
 #define HAVE_UDP_ENC_RX_OFFLOAD
 #endif /* 4.8.0 */
@@ -5556,6 +5564,7 @@
 }
 #endif
 
+#ifndef OPENSUSE_423
 static inline void __page_frag_cache_drain(struct page *page,
 					   unsigned int count)
 {
@@ -5571,6 +5580,7 @@
 #endif
 	__free_pages(page, compound_order(page));
 }
+#endif /* !OPENSUSE_423 */
 #endif /* !SLE_VERSION(12,3,0) */
 #ifndef ETH_MIN_MTU
 #define ETH_MIN_MTU 68
